pydra/engine/helpers.py

```diff
@@ -112,13 +112,27 @@ def save(task_path: Path, result=None, task=None):
         raise ValueError("Nothing to be saved")
     task_path.mkdir(parents=True, exist_ok=True)
     if result:
+        if Path(task_path).name.startswith("Workflow"):
+            # copy files to the workflow directory
+            result = copyfile_workflow(wf_path=task_path, result=result)
         with (task_path / "_result.pklz").open("wb") as fp:
             cp.dump(result, fp)
     if task:
         with (task_path / "_task.pklz").open("wb") as fp:
             cp.dump(task, fp)
 
 
+def copyfile_workflow(wf_path, result):
+    """ if file in the wf results, the file will be copied to the workflow directory"""
+    for field in attr_fields(result.output):
+        value = getattr(result.output, field.name)
+        if is_existing_file(value):
```

Comment:
this doesn't take containers (lists, tuples, dicts) into account.

Reply:
it doesn't

Reply:
i could work on this tomorrow, didn't think about it

Reply:
actually i'm not sure, let me  check